ROADMAP.md

```diff
@@ -0,0 +1,154 @@
+**Parallelism** TODO: parallel command list creation
+**Data-Driven** TODO: efficient rendering -> minimizing draw calls and overhead
+
+**Next steps**
+ * Transfer to physically based rendering (pretty large tasks..)
+ * Evaluate current system in terms of extendability for developers to implement own rendering approaches
+ * Feedback practical experience into improvements for `gfx-rs`
+
+### Input Handling
+**Modularity** The API should be flexible enough to support multiple different input devices including gamepads and maybe VR devices.
+**Data-Driven** TODO: raw input
+
+**Next steps**
+ * Raw input support (winit?)
+ * Gamepad support (winit?)
+ * Input Mapping
+
+### Audio
+TODO: Unfortunately I'm not that deep into the audio components of game engines (state of art?)
+I imagine some kind of OpenAL like API, supporting spatial sound source objects, audio streaming, attenuation, etc.
+Provide low-level access?
+
+** Next Steps**
+
+### Serialization/Reflection
+Type reflection offers additional information about the type, which is available at compile- or run-time. In comparison to C++, to can be achieved easier with Rusts macro system.
+In our engine we want to couple the ECS with serialization/reflection to easily load/save game scenes from various formats. Additionally, it helps us in to build safe network protocols, writing script and GUI bindings.
+
+** Next Steps**
+ * Integrate serde based reflection
+ * Serialize/Deserialize ecs worlds
+
+### Streaming/Asset Handling
+TODO
+
+**Parallelism** TODO: Load assets in parallel
+**Data-Driven** TODO: low loading overhead
+
+** Next Steps**
+ * Improve current approach regarding robustness (error handling) and flexibility with scene formats
+ * Implement loaders for common file formats and archives
+
+### Documentation/Tutorial
+TODO
+
+** Next Steps**
+
+### Physics
+TODO: I would consider physics pretty difficult part to do *right*, therefore not part of the short term goals.
+Maybe rely on external libraries first.
+Should still provide ways for collision detection, allowing to do collision checks, raycasts, etc. (-> ncollide?).
+
+**Parallelism** TODO: internal physic parallelism, possible GPU implementation (hard), difficult with external libraries
+**Data-Driven** TODO: integration issue: physics world <-> ECS world, having 2 different representations? if so how to keep them in sync and integrate with other conflicting processors (e.g spatial hierarchy)
+
+** Next Steps**
+ * Integrate an existing library or write from scratch (requires further planning then)
+ * Provide collision testing API (triggers, raycasts, etc.)
+
+### Tooling
+We want to focus on developing a game engine, even though a rich toolset is required to simplify different task for users. If possible, we can integrate existing tools like profilers and 3D editors.
+'In-house' tools would be implemented on demand, e.g. engine specific data formats, own editor (Qt based?).
```

Comment:
As was discussed earlier, making plugins for tools like Blender, 3DS max etc would make this process much easier for the team and the user.